tests/functional/codegen/modules/test_interface_imports.py

```diff
@@ -58,3 +58,56 @@ def foo() -> bool:
     c = get_contract(main, input_bundle=input_bundle)
 
     assert c.foo() is True
+
+
+def test_import_interface_constants(make_input_bundle, get_contract):
+    ifaces = """
+FOO: constant(uint256) = 3
+
+interface IFoo:
+    def foo() -> DynArray[uint256, FOO]: nonpayable
+    """
+
+    contract = """
+import ifaces
```

Comment:
I have a question here (without having read the full changelog). If you define a `constant` in an interface, this constant has the visibility `public` by default I think (see https://github.com/vyperlang/vyper/pull/4178) and thus if you do `implements` you must create a getter in the main contract to be compliant? It just confuses me how we handle this properly.

Reply:
I didn't consider that, but as it is at the moment, constants in an interface are not external by default`. Do we want the behaviour to be public by default?

I can see why that makes sense for consistency but IMO it would be better to not treat constants the same way. The mode of declaring a constant as public is different from that for a function, and also we would not expect the getter of a constant to be part of the interface if it was not explicitly declared as public.

Reply:
Well, that's exactly why I raise this point to highlight the inconsistency here. And I agree with you that it should not be handled the same way, but given the current semantics, it could be interpreted like this. Also, that's why I think having _optional_ `external` visibility in interfaces is not appropriate really; explicitness is better.

Reply:
yes, that is a good point. the idea here is that constants are like types, they can be imported for use in your contract, but they shouldn't be allowed to be marked `external`

Reply:
can we maybe make this more clear via some syntax sugar? An interface, in the context of smart contract programming, signals something "external", and given that now the visibility modifier is even only optional, we should make the scope clear for `constant` values to avoid any ambiguity.

Reply:
Well I think `constant`s can be useful for this type definitions, and since it's frequently requested we should listen and ship this feature. How about something like that:

```vy
FOO: constant(uint256) = Literal[3]
```

Reply:
Maybe we can also consider (1) throwing an exception if a constant is declared as public in an interface; and/or (2) throwing a warning that constants in interfaces are not and cannot be public.

Reply:
Ok another suggestion: how about this in the interface?

```vy
@internal
    FOO: constant(uint256) = 3
    BAR: constant(bytes1) = 0xff
    ...
```

Reply:
That feels a little hacky. 

How about we skip this for `0.4.1` and introduce type aliases?

something in this spirit:
```python
NUM_ITEMS: alias(uint256) = 5
arr: alias(DynArray[uint256, NUM_ITEMS])

def foo(a: arr):
    pass
```

Reply:
in the past, Charles suggested that aliases might also help with this (a bit ugly 😉 ) interface syntax
https://github.com/vyperlang/vyper/pull/4090#issuecomment-2351171352

Reply:
i mean i think if you can use it in type declarations, you should be able to use it like any other constant. e.g.
```
for i: uint256 in range(MyInterface.NUM_ITEMS):
    ...
```

Reply:
Based on offline discussion I think we shouldkeep the changes for flags herr, roll back the changes for constants here and punt to future work / a separate PR

Reply:
sgtm